subsys/bluetooth/mesh/shell/shell.c

```diff
@@ -138,9 +139,14 @@ static const struct bt_mesh_health_srv_cb health_srv_cb = {
 	.attn_on = attention_on,
 	.attn_off = attention_off,
 };
+#endif /* CONFIG_BT_MESH_SHELL_HEALTH_SRV */
+
+
 
 struct bt_mesh_health_srv bt_mesh_shell_health_srv = {
```

Comment:
I'm bit confused because of the description of the PR. Shouldn't the 'shell.c' automatically pick up the Health Server instance from the application? Because Health Server is always mandatory in all mesh applications. That should technically allow shell.c to fetch the instance and then redirect the original callback (if any defined) by using its own callback.

I think this instantiation should not be here, but rather the instance is fetched from the application using access APIs.

Reply:
Ok, so you want the shell to overwrite the application CB's with an internal shell callback which again calls the application CB? In that case this was suggested for the prov improvments for shell, but @alxelax pointed out some risks/issues by doing so: https://github.com/zephyrproject-rtos/zephyr/pull/50180#pullrequestreview-1105635127. Another challenge is that there can be several instances of the health server as defined by the profile spec. Should the shell then only overwrite the root element health server, or should it also interfer with any secondary instances?

Reply:
I think to get full transparency of the shell, the shell can override all callback instances. That way shell can intercept the callbacks, do its thing (print or update internal states) and pass on the info to the application. 

This will enable application fault logic testing, if any, implemented in the application and become a handy tool.